resources/js/components/ui/calendar/FullCalendar.tsx

```diff
@@ -0,0 +1,266 @@
+import {Capitalize} from "../Capitalize";
+import {SegmentedControl} from "../form";
+import {ResourceGroup, ResourcesSelector} from "./ResourcesSelector";
+import {TinyCalendar} from "./TinyCalendar";
+import {DaysRange} from "./days_range";
+import {WeekDaysCalculator} from "./week_days_calculator";
+
+export type Mode = "month" | "week" | "day";
+
+interface Props extends htmlAttributes.div {
+  locale: Intl.Locale;
+  resourceGroups: readonly ResourceGroup[];
+  holidays?: readonly DateTime[];
+  modes?: Mode[];
+  initialMode?: Mode;
+  initialResourcesSelection?: readonly string[];
+  initialDay?: DateTime;
+}
+
+const defaultProps = () =>
+  ({
+    modes: ["month", "week", "day"],
+    initialMode: "week",
+    initialResourcesSelection: [] as readonly string[],
+    initialDay: currentDate(),
+  }) as const;
+
+/**
+ * A full-page calendar, consisting of a tiny calendar, a list of resources (people), calendar mode
+ * switcher, and a large calendar with either month view, or hours view.
+ */
+export const FullCalendar: VoidComponent<Props> = (propsArg) => {
+  const mProps = mergeProps(defaultProps(), propsArg);
+  const [props, divProps] = splitProps(mProps, [
+    "locale",
+    "resourceGroups",
+    "holidays",
+    "modes",
+    "initialMode",
+    "initialResourcesSelection",
+    "initialDay",
+  ]);
+
+  const t = useLangFunc();
+  const [mode, setMode] = createSignal(props.initialMode);
+  const weekDayCalculator = createMemo(() => new WeekDaysCalculator(props.locale));
+  function getRange(day: DateTime) {
+    const m = mode();
+    switch (m) {
+      case "month":
+        return new DaysRange(day.startOf("month"), day.endOf("month"));
+      case "week":
+        return weekDayCalculator().dayToWorkdays(day);
+      case "day":
+        return DaysRange.oneDay(day);
+      default:
+        return m satisfies never;
+    }
+  }
+  const [tinyCalMonth, setTinyCalMonth] = createSignal(props.initialDay);
+  // Initialise to whatever range, it will be overwritten below.
+  const [daysSelection, daysSelectionSetter] = createSignal(DaysRange.oneDay(props.initialDay));
+  // Initialise to whatever range, it will be immediately updated by the calendar.
+  const [tinyCalVisibleRange, setTinyCalVisibleRange] = createSignal(DaysRange.oneDay(props.initialDay));
+  // Initialise to empty, it will be overwritten below.
+  const [selectedResources, selectedResourcesSetter] = createSignal<string[]>([]);
+
+  /** The resources selection view, allowing multiple selection only in the day mode. */
+  const resourcesSelectionMode = () => (mode() === "day" ? "checkbox" : "radio");
+  /** The last resources selection for each resources selection mode. */
+  const resourcesSelectionByMode = {
+    checkbox: props.initialResourcesSelection.slice(),
+    radio: props.initialResourcesSelection.slice(0, 1),
+  };
+  // eslint-disable-next-line solid/reactivity
+  selectedResourcesSetter(resourcesSelectionByMode[resourcesSelectionMode()]);
```

Comment:
W tych selected resources to trochę trudno się było połapać. Zastanawiam się czy nie było by trochę prościej gdyby to checkbox i radio były sygnałami:

```js
const [selectedResourcesCheckbox, selectedResourcesCheckbox] = createSignal(initialResources);
const [selectedResourceRadio, setSelectedResourceRadio] = createSignal(initialResources[0]);
const selectedResources = createMemo(() => resourcesSelectionMode() === 'checkbox' ? selectedResourcesCheckbox() : [selectedResourceRadio]);
function setSelectedResources(ids) {
  if (resourcesSelectionMode() === 'checkbox') {
    setSelectedResourcesCheckbox(ids);
  } else if (resourcesSelectionMode() === 'radio') {
    setSelectedResourcesRadio(ids[0]);
  }
}
```

I jeszcze trochę kodu restore zostanie.

Nie wiem czy to lepiej będzie, do rozważenia, ale na oko mi się wydało czyściej.

Reply:
Coś tam wyklepałem, być może czyściej. Pytanie zostaje czy podobna zamiana daysSelection też by miała sens, czy jednak nie. Na razie w każdym razie nie robię.